src/components/SignInButton.tsx

```diff
@@ -73,22 +77,28 @@ export const SignInButton = (props: SignInButtonProps) => {
   // Check for ToS Acceptance - redirect user if not set.
   const checkToSAndRedirect = async (redirectPath: string | null) => {
     // Check if the user has accepted ToS yet or not:
-    const user = await User.getMe();
+    const user: DuosUserResponse = await User.getMe();
     if (!user.roles) {
       await StackdriverReporter.report('roles not found for user: ' + user.email);
     }
     setUserRoleStatuses(user, Storage);
     await onSignIn();
     const userStatus = await ToS.getStatus();
-    const { tosAccepted } = userStatus;
+    const {tosAccepted} = userStatus;
     if (!isEmpty(userStatus) && !tosAccepted) {
-      await Auth.signOut();
+      // User has authenticated, but has not accepted ToS
+      Storage.setUserIsLogged(false);
```

Comment:
This is the fix that allows for an authenticated user to access `/tos_acceptance`